discover/deployment/articles/06-upgrading-liferay/03-running-the-upgrade-process.markdown

```diff
@@ -347,7 +358,12 @@ module's upgrade status. The core's `servletContextName` field value is
 Each module has one `Release_` table record, and the value for its
 `schemaVersion` field must `1.0.0` or greater (`1.0.0` is the initial version
 for @product-ver@ modules, except for those that were previously traditional
-plugins intended for Liferay Portal version 6.2 or earlier). 
+plugins intended for Liferay Portal version 6.2 or earlier).
+
+Apart from the dependencies with other modules, modules can also have other
+dependencies such as having a certain schema version in the `Release_` table.
+You can check why some modules are not available even when they are actived
+executing `dm wtf`. This logic is part of [Apache Felix Dependency Manager](http://felix.apache.org/documentation/subprojects/apache-felix-dependency-manager/tutorials/leveraging-the-shell.html).
```

Comment:
@achaparro Can you provide an example situation where a module would depend on a certain version the Release_ table and why that version wouldn't already be in the table? Thanks.

Reply:
@jhinkey This happens when we establish a certain require schemaVersion like this:
https://github.com/liferay/liferay-portal/blob/master/modules/apps/web-experience/journal/journal-service/bnd.bnd#L23

The module won't be available until the required upgrade processes have been executed and the schemaVersion matches with the requireSchemaVersion, so if the upgrade process is still being executing or there are some errors in the portlet definition, we could check why it is not availabe yet with that command. It will show the unfulfilled dependencies in this way:
1 missing dependencies found.
-------------------------------------
The following service(s) are missing:
 * com.liferay.portal.kernel.model.Release (&(release.bundle.symbolic.name=com.liferay.journal.service)(release.schema.version=1.0.2)) is not found in the service registry

The upgrade:check shows something similar but "dm wtf" can also help to detect errors defining schemaVersion fields during the creation of custom portlets.

Please, let me know if you need something else.

Thanks.